docs/assemblies/assembly-record-validation-filter.adoc

```diff
@@ -0,0 +1,27 @@
+// file included in the following:
+//
+// assembly-built-in-filters.adoc
+
+[id='assembly-record-validation-filter-{context}']
+= (Preview) Record Validation
+
+[role="_abstract"]
+The Record Validation filter validates records sent by a producer.
+Only records that pass the validation are sent to the broker.
+This filter can be used to prevent _poison messages_—such as those containing corrupted data or invalid formats—from entering the Kafka system, which may otherwise lead to consumer failure.
+
+The filter currently supports two modes of operation:
+
+1. Schema validation validates the content of the record against a schema stored in an https://www.apicur.io/registry/[Apicurio Registry].
+2. JSON Syntax validation ensures the producer is producing messages that contain syntactically valid JSON.
```

Comment:
```suggestion
1. Schema validation mode validates the content of the record against a schema stored in an https://www.apicur.io/registry/[Apicurio Registry].
2. JSON Syntax validation mode ensures the producer is producing messages that contain syntactically valid JSON.
```

Reply:
I rephrased in another way, please take a look.